client/src/components/TaskCard/TaskCard.js

```diff
@@ -35,9 +35,15 @@ const TaskCard = ({
   };
 
   const reflectionDoneHandler = async (reflectionComments, satisfaction) => {
-    await createTaskReflectionHandler(id, reflectionComments, satisfaction);
-    await toggleTaskHandler(id);
-    toggleReflectionModal();
+    const success = await createTaskReflectionHandler(
```

Comment:
I think we can remove `await` here

Reply:
I tested this on my end and I think we still need the `await` here.

There are two reasons:

- we want to ensure `createTaskReflectionHandler` completes first before we execute `toggleTaskHandler`. 
- without the `await`, `success` would be a `Promise` instead of a `bool`. Since non-null objects evaluate to `true` in JS, this would cause the if-statement to always evaluate to `true`, beating the point of the check. (I suspect this is why it looked like it ran find when you removed `await`).

Even though the enclosing function is `async`, we still need to use `await` to ensure the calls run synchronously. Below, I've attached the console log for with and without `await`.

With `await`:
![image](https://user-images.githubusercontent.com/69379333/165661123-9691947a-7402-4512-87a3-892d0f68f6b5.png)

Without `await`:
![image](https://user-images.githubusercontent.com/69379333/165661149-12f3fe79-40eb-405f-b0bf-d92b41fbda71.png)

`TaskCard.js`:
![image](https://user-images.githubusercontent.com/69379333/165661212-fb6c10e4-2dab-4e77-8f4f-5b5fca3b901d.png)

Reply:
Gotcha yep, that definitely makes sense! Good catch!